### DISCUSSION TITLE
Подскажите нубу

### DISCUSSION THREAD

**Discussion by bigmaniak** [Anonym-tsk/nfqws-keenetic#21] [Q&A] [answered]
имеется FT-AIR-DUO-G
инструкция из 2 строчек (автоматическая установка), но не могу ничего установить
выдает такое
![genb](https://github.com/user-attachments/assets/a08aea75-174e-4e44-8f27-e387564be789)
Что я делаю не так?

**Comment by Anonym-tsk** [comment, marked as answer]
Попробуйте opkg install busybox

**Reply by bigmaniak** [reply]
лучи добра тебе в карму, добрый человек!

**Comment by bigmaniak** [comment]
Новый вопрос)))
всё установилось, ребутнул роутер. Он влючился, интернет появился. Но теперь не заходит на вебморду и по SSH...
Общий сброс роутера поможет вернуть всё назад?

**Reply by Anonym-tsk** [reply]
Общий сброс то всегда поможет. Но вообще, вряд ли проблема как-то связана с nfqws. Попробуйте телнет, если он там есть.

**Reply by bigmaniak** [reply]
не подключается
![telnet](https://github.com/user-attachments/assets/87cad3a6-ab2d-4c03-8e4a-491ae5662f65)
при этом интернет, wi-fi есть...роутер свои функции выполняет, но не пускает к себе...

**Reply by Anonym-tsk** [reply]
Ну я даже не знаю что посоветовать. Другой браузер? С другого устройства зайти? Веб морда то должна работать

**Reply by bigmaniak** [reply]
ладно...буду разбиратся. Пара вопросов осталась
Если вытащить флешку из роутера - скрипт остановится/не запустится?
если без перезагрузки встаить флешку - скрипт сам запустится?

**Reply by Anonym-tsk** [reply]
Если на флешке entware, то без флешки он не запустится.
Если вставить флешку без перезагрузки - не должен запуститься сам

**Reply by bigmaniak** [reply]
и еще один вопрос, как понять что скрип выполняется? в логах роутера увижу его запуск?
вот кусочек лога роутера после перезагрузки:

Aug 18 23:50:42 automount: unknow fs type on /dev/sda2 try mount with default options for NTFS
Aug 18 23:50:42 automount: add /dev/sda2 with ntfs
Aug 18 23:50:42 automount: can not mount ntfs /dev/sda2 /media/sda2
Aug 18 23:50:43 automount: add /dev/sda5 with ext4
Aug 18 23:50:43 automount: detect optware part on sda5
Aug 18 23:50:43 kernel: EXT4-fs (sda5): recovery complete
Aug 18 23:50:43 kernel: EXT4-fs (sda5): mounted filesystem with ordered data mode. Opts: (null)
Aug 18 18:50:43 automount: start Entware services...
Aug 18 23:50:43 kernel: xt_connbytes: Forcing CT accounting to be enabled
Aug 18 23:51:03 adblock: Filter records.
Aug 18 23:51:08 kernel: ASSOC - Assign VHT STA MODE=VHT, MCS=9, BW=80M, AID=4 to 5GHz AP aa:b2:92:51:33:dd, WNM supported, PMF connect (WPA2PSK/AES)
Aug 18 23:51:09 udhcpd[3683]: sending OFFER to 192.168.1.36
Aug 18 23:51:09 udhcpd[3683]: sending ACK to 192.168.1.36
Aug 18 23:51:19 adblock: Remove duplicated records.
Aug 18 23:51:49 dropbear[6354]: Child connection from 192.168.1.132:58960
Aug 18 23:51:55 adblock: 162054 domains blocked by DNS.
Aug 18 23:51:55 dnsserver: Stopping DNSMASQ.
Aug 18 23:51:55 dnsmasq[5140]: exiting on receipt of SIGTERM
Aug 18 23:51:56 dnsserver: Generate /etc/hosts file.
Aug 18 23:51:56 dnsserver: Starting DNSMASQ.
Aug 18 23:51:56 adblock: Next adblock update after 24h.
Aug 18 23:51:56 dropbear[6354]: Password auth succeeded for 'Admin' from 192.168.1.132:58960
Aug 18 23:51:57 dnsmasq[6449]: started, version 2.8x cachesize 4096
Aug 18 23:51:57 dnsmasq[6449]: compile time options: IPv6 GNU-getopt no-RTC no-DBus no-UBus no-i18n no-IDN no-DHCP no-scripts no-TFTP no-conntrack no-ipset no-nftset no-auth cryptohash DNSSEC loop-detect inotify no-dumpfile
Aug 18 23:51:57 dnsmasq[6449]: using 162023 more nameservers
Aug 18 23:52:00 dnsmasq[6449]: read /etc/hosts - 5 names
Aug 18 23:52:00 dnsmasq[6449]: using nameserver 10.200.201.162#53
Aug 18 23:52:00 dnsmasq[6449]: using nameserver 10.200.201.234#53
Aug 18 23:52:00 dnsmasq[6449]: using 162023 more nameservers
Aug 18 23:52:20 kernel: ASSOC - Assign VHT STA MODE=VHT, MCS=25, BW=80M, AID=5 to 5GHz AP 06:18:93:21:4f:6d, WNM supported, PMF connect (WPA2PSK/AES)
Aug 18 23:52:21 udhcpd[3683]: sending OFFER to 192.168.1.238
Aug 18 23:52:21 udhcpd[3683]: sending ACK to 192.168.1.238

или в терминале какой-то командой можно понять что скрипт выполняется?
при автоматической установке на флешку ставится? там где /opt

**Reply by Anonym-tsk** [reply]
В ридми внизу есть команды

**Reply by arcanium1** [reply]
А можно и мне нубский вопрос задать? 

После ввода 
/bin/sh -c "$(curl -fsSL https://github.com/Anonym-tsk/nfqws-keenetic/raw/master/netinstall.sh)"
Получаю:
curl: error while loading shared libraries: libssl.so.3: cannot open shared object file: No such file or directory

Команды ручной установки работают, но ютуб так и не завелся...
В конце ручной установки

/opt/etc/init.d/S51nfqws: line 41: /opt/usr/bin/nfqws: not found
Started NFQWS service
Installation successful

**Reply by bigmaniak** [reply]
> В ридми внизу есть команды

спс, нашел...поигрался командами, вот что выдает
![cnenec](https://github.com/user-attachments/assets/d2e3cab4-e304-45f1-96c5-3a5d7043d2b1)
я так понял скрипт не запускается...

**Reply by Anonym-tsk** [reply]
> А можно и мне нубский вопрос задать?

Давайте в отдельной ветке. Опишите что за железо у вас и какая прошивка

**Reply by Anonym-tsk** [reply]
> я так понял скрипт не запускается

Бинарник не скачался. Переустановите скрипты заново и покажите вывод

**Reply by bigmaniak** [reply]
> > я так понял скрипт не запускается
> 
> Бинарник не скачался. Переустановите скрипты заново и покажите вывод

[Wive-NG-HQ:/home]# /bin/sh -c "$(curl -fsSL https://github.com/Anonym-tsk/nfqws
-keenetic/raw/master/netinstall.sh)"
######################################################################### 100.0%
######################################################################################################################################################################### 100.0%
######################################################################################################################################################################### 100.0%
######################################################################################################################################################################### 100.0%
-=O#-   #      #       #
######################################################################################################################################################################### 100.0%
######################################################################################################################################################################### 100.0%
######################################################################################################################################################################### 100.0%

Begin install? y/N
y
Downloading http://bin.entware.net/mipselsf-k3.4/Packages.gz
Updated list of available packages in /opt/var/opkg-lists/entware
Package iptables (1.4.21-4) installed in root is up to date.
cp: can't create '/opt/etc/ndm/netfilter.d/100-nfqws.sh': No such file or directory
chmod: /opt/etc/ndm/netfilter.d/100-nfqws.sh: No such file or directory

Select the router architecture: mipsel (default), mips, aarch64, arm
  mipsel  - KN-1010/1011, KN-1810, KN-1910/1912, KN-2310, KN-2311, KN-2610, KN-2910, KN-3810
  mips    - KN-2410, KN-2510, KN-2010, KN-2012, KN-2110, KN-2112, KN-3610
  aarch64 - KN-2710, KN-1811
  arm     - Other arm-based devices
mipsel
Selected architecture: mipsel
Warning: Failed to open the file /opt/usr/bin/nfqws: No such file or directory
curl: (23) client returned ERROR on write of 1378 bytes

chmod: /opt/usr/bin/nfqws: No such file or directory

----------------------
lo 127.0.0.1
eth3 10.188.214.43
br0 192.168.1.1

Enter the provider interface name from the list above, e.g. eth3 (default) or nwg1
You can specify multiple interfaces separated by space, e.g. eth3 nwg1
eth3
Selected interface: eth3

Select working mode: auto (default), list, all
  auto - automatically detects blocked resources and adds them to the list
  list - applies rules only to domains in the list /opt/etc/nfqws/user.list
  all  - applies rules to all traffic except domains from list /opt/etc/nfqws/exclude.list
auto
Selected mode: auto

Disable IPv6 support (enabled by default)? y/N
y
xt_multiport.ko loaded
xt_connbytes.ko loaded
xt_NFQUEUE.ko loaded
/opt/etc/init.d/S51nfqws: line 41: /opt/usr/bin/nfqws: not found
iptables: No chain/target/match by that name.
iptables: No chain/target/match by that name.
iptables: No chain/target/match by that name.
Started NFQWS service
Installation successful
[Wive-NG-HQ:/home]#

**Reply by Anonym-tsk** [reply]
Кажется у вас система не в /opt

**Reply by bigmaniak** [reply]
Какая система?

**Reply by Anonym-tsk** [reply]
> Какая система?

entware

**Reply by Anonym-tsk** [reply]
Покажите
```
ls -la /opt
```

**Reply by bigmaniak** [reply]
> Покажите
> 
> ```
> ls -la /opt
> ```

![mount](https://github.com/user-attachments/assets/d07d0ebf-c6e4-4c4a-98a3-93bc8daeeefd)
![ls](https://github.com/user-attachments/assets/d66dfa82-8724-4fed-8d95-58cae6fd440d)

**Reply by Anonym-tsk** [reply]
Ну вот странно, а если `ls -la /opt/usr`

**Reply by bigmaniak** [reply]
![ls2](https://github.com/user-attachments/assets/ff6e9bca-1513-4a7a-9fe2-dde3888bddac)

**Reply by Anonym-tsk** [reply]
Странная железяка. Создайте руками каталог /opt/usr/bin, после этого переустановите скрипты

**Reply by bigmaniak** [reply]
Всё равно не стартует, хотя ошибок стало меньше
![ins](https://github.com/user-attachments/assets/63994e15-0365-45fb-b282-4f7747b9eaf6)

**Reply by Anonym-tsk** [reply]
Что за прошивка? Похоже там nfqueue отсутствует

**Reply by bigmaniak** [reply]
прошивка стоковая, Wive NG, версия 6.5.9.RU.21042023
пойду на 4PDA, выяснять насчет nfqueue

**Reply by bigmaniak** [reply]
iptables v1.4.16.3: Couldn't find target `NFQUEUE'
Нету...
Его можно же доставить из ethware?

**Reply by bigmaniak** [reply]
Доброго дня! вопрос остался открытым
Было бы неплохо, если бы кто-нибудь с Keenetic такой же архитектуры mipsel вытащил нужные модули...netfilter. Изначально их в кинетике же нет, они доустанавливаются же....

**Comment by walkman99** [comment]
Возможно странный вопрос, но как обновлять скрипт, если была автоматическая установка?
